x/batch/msg_test.go

```diff
@@ -19,17 +20,14 @@ func TestMsg(t *testing.T) {
 			msg.AssertExpectations(t)
 		})
 
-		Convey("Test list too long", func() {
-			msg.On("MsgList").Return(make([]weave.Msg, 11), nil)
-			So(batch.Validate(msg), ShouldNotBeNil)
+		Convey("Test validation errors", func() {
+			msg.On("MsgList").Return(make([]weave.Msg, 11), errors.ErrEmpty)
+			err := batch.Validate(msg).(errors.Multi)
```

Comment:
This can cause panic. The below tests show what I have in mind

https://github.com/iov-one/weave/blob/eaf86c92555a8b2cca3e6755a3d36231944fd4fc/errors/multi_test.go#L7-L16

What I do is to call some `validate` function that can return an `error` instance. This `validate` function is successful and returns `nil`.
I am cassting the result to `errors.Multi`, because I want to run some tests. This code panics, because `error` does not implement `errors.Multi`.

Reply:
That's a good catch, I have fixed it. 

Although since we were looking at using this only for tests - you'd expect to know what is being returned when. But the new api is nicer